src/surf_collide_specular.cpp

```diff
@@ -27,7 +27,19 @@ using namespace SPARTA_NS;
 SurfCollideSpecular::SurfCollideSpecular(SPARTA *sparta, int narg, char **arg) :
   SurfCollide(sparta, narg, arg)
 {
-  if (narg != 2) error->all(FLERR,"Illegal surf_collide specular command");
+  if (narg < 2 || narg > 3) error->all(FLERR,"Illegal surf_collide specular command");
+
+  // optional args
+
+  adiabatic_flag = 0;
+
+  int iarg = 2;
+  while (iarg < narg) {
+    if (strcmp(arg[iarg],"adiabatic") == 0) {
+      adiabatic_flag = 1;
+      iarg += 1;
+    }
```

Comment:
this needs an error test in case adiabatic is misspelled or a wrong keyword is used

Reply:
Good point, will fix